server/lib/history.js

```diff
@@ -0,0 +1,31 @@
+const _ = require('lodash');
+
+module.exports = (record, storage) =>
+  storage.read()
+    .then(data => {
+      data = data || {};
+      data.jobs = data.jobs || [];
+      const job = {
+        id: record.id,
+        type: record.type,
+        date: record.date || new Date(),
+        status: record.status || 'pending',
+        summary: record.summary || {}
+      };
+
+      const jobIndex = _.findIndex(data.jobs, { id: job.id });
+
+      if (jobIndex < 0) {
+        data.jobs.push(job);
+      } else {
+        data.jobs[jobIndex].status = job.status;
+        data.jobs[jobIndex].summary = job.summary;
+      }
+
+      if (data.jobs.length > 20) {
```

Comment:
how much data per job?  Could we do more?  Should we be checking by size of data instead in case some jobs are bigger than others?

Reply:
```
"id": "job_HaANV9IdXfr4ngIS",
"type": "import",
"date": "2018-02-06T10:14:06.141Z",
"status": "completed",
"summary": {
   "failed": 3,
   "updated": 0,
   "inserted": 0,
   "total": 3
}
```

Not much data there. We're doing `/api/v2/jobs/${id}/errors` requests for getting report, and don't store it (because there could be too much data). Also, jobs expires rather quickly, so i don't think we need to store more then last 20.

Reply:
Is there some way to limit the number of records returned from `storage.read()` instead of pruning the list after the fact? Or is there a good reason not to do that (maybe the particular job we're looking for would not be returned)?